packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -329,7 +329,7 @@ export class IndexedDbPersistence implements Persistence {
       })
       .then(() => {
         return this.simpleDb.runTransaction(
-          'readonly',
+          'readonly-idempotent',
```

Comment:
This appears to modify `this.listenSequence` in the transaction.

Why not move that out of the transaction and into the first `then` block below?

Reply:
There are a bunch of calls in the SDK that follow this pattern and re-compute a value inside a transaction, but the re-computation has no side effects. In this case, refactoring the change out of this transaction is pretty simple, so I applied the change. It might not always be so simple in other cases though.